### DISCUSSION TITLE
Riverpod dependencies lint and Service local variable

### DISCUSSION THREAD

**Discussion by rsegecin** [rrousselGit/riverpod#3503] [Q&A] [answered]
Hi, I'm getting two kind of lint warnings and I'm trying to get rid of them. 

The first is on the declaration of dependency Services that I think I'm doing it right but it doesn't go away it says `If a provider depends on providers which specify "dependencies", they should themselves specify "dependencies" and include all the scoped providers.`, as you can see both of the riverpod classes are generated and `ServiceTest` just depends on `ServicePrivateNumber` so why am I still getting this warning? 

The second is a local variable that I'd like to have along with the state that I'm holding in my service. I understand that I should have a method to access and modify the Service's state, but I think it's too much to create another service just to hold a single local scope variable, see example below.

```
import 'package:riverpod_annotation/riverpod_annotation.dart';

part 'service_test.g.dart';

@Riverpod()
class ServicePrivateNumber extends _$ServicePrivateNumber {
  @override
  int build(int number) {
    return number + 1;
  }
}

@Riverpod(dependencies: [ServicePrivateNumber])
class ServiceTest extends _$ServiceTest {
  int privateNumber = 0;

  @override
  String build(int number) {
    ref.watch(servicePrivateNumberProvider(number * 10));
    return number.toString();
  }

  int getPrivateNumber() {
    return ref.read(servicePrivateNumberProvider(number * 10));
  }
}
```

![image](https://github.com/rrousselGit/riverpod/assets/4460859/b504d407-a905-4840-9ef8-0da07029ed2a)

**Comment by snapsl** [comment, marked as answer]
1. ServicePrivateNumber does not define dependencies as the error suggests. In your case it's likely an empty list.
`@Riverpod(dependencies: [])`
2. The issue with a local variable in a notifier is that your state is not affected when modifying the local variable. (`ref.watch()` won't trigger) 
If that is what you want, go for it. Even though, it can be seen as an anti-pattern. Writing a getter is more desirable.

**Reply by rsegecin** [reply]
1. `@Riverpod(dependencies: [])` did the trick
2. In my use case changing the local variable doesn't need to trigger the Service's notifier. I don't like doing things anti pattern but I'm still a human being. I think in this case its not a cardinal sin is it?

Thank you @snapsl.

**Comment by rrousselGit** [comment]
There's an Dart bug that can cause lints to be out of date in the IDE. If you refresh, do you still see it?

**Reply by rsegecin** [reply]
Even after refreshing or reopening the project I was still able to see it. @snapsl's made them go away. The local variable I'm a bit skeptical by being "anti pattern" but I think I'll do it anyway. I'd like to hear your opinion about it but in any case thanks.

Thank you @rrousselGit.

**Reply by rrousselGit** [reply]
Oh yes my bad. That's because your ServicePrivateNumber does not specify `dependencies: []`, so specifying it is useless.
